aether-kernel/aether/kernel/api/migrations/0016_auto_20180904_1429.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.15 on 2018-09-04 14:29
+from __future__ import unicode_literals
+
+import django.contrib.postgres.fields.jsonb
+from django.db import migrations, models
+import django.db.models.deletion
+import django.utils.timezone
+import django_prometheus.models
+import model_utils.fields
+import uuid
+
+
+def migrate_current_mappings_to_mappingsets(apps, schema_editor):
+    Mapping = apps.get_model('kernel', 'Mapping')
+    MappingSet = apps.get_model('kernel', 'MappingSet')
+    Submission = apps.get_model('kernel', 'Submission')
+
+    for mapping in Mapping.objects.all():
+        mappingset = MappingSet.objects.create(
+            pk=mapping.pk,
+            name=mapping.name,
+            project=mapping.project,
+        )
+        mapping.mappingset = mappingset
+        mapping.save()
+
+        submissions_by_mapping = Submission.objects.filter(mapping=mapping)
+        for submission in submissions_by_mapping:
+            submission.mappingset = mappingset
+            submission.save()
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('kernel', '0015_auto_20180725_1310'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='MappingSet',
+            fields=[
+                ('created', model_utils.fields.AutoCreatedField(default=django.utils.timezone.now, editable=False, verbose_name='created')),
+                ('modified', model_utils.fields.AutoLastModifiedField(default=django.utils.timezone.now, editable=False, verbose_name='modified')),
+                ('id', models.UUIDField(default=uuid.uuid4, primary_key=True, serialize=False)),
+                ('revision', models.TextField(default='1')),
+                ('name', models.CharField(max_length=50, unique=True)),
+                ('input', django.contrib.postgres.fields.jsonb.JSONField(blank=True, null=True)),
+            ],
+            options={
+                'ordering': ['project__id', '-modified'],
+                'default_related_name': 'mappingsets',
+            },
+        ),
+        migrations.AlterModelOptions(
+            name='mapping',
+            options={'ordering': ['-modified']},
+        ),
+        migrations.RemoveIndex(
+            model_name='mapping',
+            name='kernel_mapp_project_0cb3fd_idx',
+        ),
```

Comment:
You can delete these two operations because you kept the Mapping indices and ordering.

```python
        migrations.AlterModelOptions(
            name='mapping',
            options={'ordering': ['-modified']},
        ),
        migrations.RemoveIndex(
            model_name='mapping',
            name='kernel_mapp_project_0cb3fd_idx',
        ),
```